packages/shared/lib/services/sync/run.service.integration.test.ts

```diff
@@ -0,0 +1,356 @@
+import { expect, describe, it, beforeAll, afterAll, vi } from 'vitest';
+import db from '../../db/database.js';
+import SyncRun from './run.service.js';
+import environmentService from '../environment.service.js';
+import LocalFileService from '../file/local.service.js';
+import { SyncStatus, SyncType } from '../../models/Sync.js';
+import * as database from '../../db/database.js';
+import * as dataMocks from './data/mocks.js';
+import * as dataService from './data/data.service.js';
+import * as recordsService from './data/records.service.js';
+import * as jobService from './job.service.js';
+import * as configService from './config/config.service.js';
+import type { IntegrationServiceInterface, Sync, Job as SyncJob, SyncResult } from '../../models/Sync.js';
+import type { Environment } from '../../models/Environment.js';
+import type { DataResponse } from '../../models/Data.js';
+import type { Connection } from '../../models/Connection.js';
+
+class integrationServiceMock implements IntegrationServiceInterface {
+    async runScript() {
+        return {
+            success: true
+        };
+    }
+    async cancelScript() {
+        return;
+    }
+}
+
+const integrationService = new integrationServiceMock();
+
+describe('Running sync', () => {
+    beforeAll(async () => {
+        await database.multipleMigrations();
+    });
+
+    afterAll(async () => {
+        await clearDb();
+    });
+
+    describe(`with track_deletes=false`, () => {
+        const trackDeletes = false;
+        it(`should report no records have changed`, async () => {
+            const rawRecords = [
+                { id: '1', name: 'a' },
+                { id: '2', name: 'b' }
+            ];
+            const expectedResult = { added: 0, updated: 0, deleted: 0 };
+            const records = await verifySyncRun(rawRecords, rawRecords, trackDeletes, expectedResult);
+            records.forEach((record) => {
+                expect(record._nango_metadata.first_seen_at).toEqual(record._nango_metadata.last_modified_at);
+                expect(record._nango_metadata.deleted_at).toBeNull();
+                expect(record._nango_metadata.last_action).toEqual('ADDED');
+            });
+        });
+
+        it(`should report one record has been added and one modified`, async () => {
+            const rawRecords = [
+                { id: '1', name: 'a' },
+                { id: '2', name: 'b' }
+            ];
+            const newRecords = [
+                { id: '1', name: 'A' },
+                { id: '3', name: 'c' }
+            ];
+            const expectedResult = { added: 1, updated: 1, deleted: 0 };
+            const records = await verifySyncRun(rawRecords, newRecords, trackDeletes, expectedResult);
+
+            records
+                .filter((record) => record.id == 1)
+                .forEach((record) => {
```

Comment:
Note that with this method if `records` is empty or not returning the correct set of records no `expect` will fail